src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -545,22 +545,18 @@ private static ISettings LoadSettingsGivenSettingsFiles(List<SettingsFile> setti
                     yield break;
                 }
 
-                // If the default user config NuGet.Config does not exist, we need to create it.
                 var defaultSettingsFilePath = Path.Combine(userSettingsDir, DefaultSettingsFileName);
 
-                SettingsFile userSpecificSettings = ReadSettings(rootDirectory, defaultSettingsFilePath, settingsLoadingContext: settingsLoadingContext);
-                if (File.Exists(defaultSettingsFilePath) && userSpecificSettings.IsEmpty())
-                {
-                    var trackFilePath = Path.Combine(Path.GetDirectoryName(defaultSettingsFilePath), NuGetConstants.AddV3TrackFile);
+                var defaultSettingsFilePathExistsPreviously = File.Exists(defaultSettingsFilePath);
 
-                    if (!File.Exists(trackFilePath))
-                    {
-                        File.Create(trackFilePath).Dispose();
+                // If the default user config NuGet.Config does not exist, we need to create an empty one.
```

Comment:
```suggestion
                // ReadSettings will try to create the config file if it doesn't exist, which is why we should check if it exists again afterwards
```

The comment "create if does not exist", does not make a lot of sense when the code below it calls a method named `ReadSettings`, and doesn't address Erick's original question.

Reply:
Thanks. Fixed~